test/ci/test-fsim-wget.sh

```diff
@@ -88,15 +86,66 @@ run_test () {
 
   echo "⭐ Prepare the wget payload on server side: '${wget_test_file}'"
   prepare_payload "${wget_test_file}"
+}
+
+# Basic file transfer test
+fsim_wget_test() {
+  fsim_wget_setup
 
   echo "⭐ Running FIDO Device Onboard with FSIM fdo.wget"
   run_fido_device_onboard --debug --wget-dir "${wget_device_download_dir}"
 
   echo "⭐ Verify downloaded file: server: ${wget_test_file} device: ${wget_device_test_file}"
   verify_equal_files "${wget_test_file}" "${wget_device_test_file}"
+  echo "⭐ Success! ✅"
+  cleanup
+}
+
+# Test failure to contact HTTP server
+fsim_wget_nohost_test() {
+  fsim_wget_setup
 
+  echo "⭐ Attempt WGET with missing HTTP server, verify FSIM error occurs"
+  stop_service "${wget_httpd_service_name}"
+
+  onboarding_log="${logs_dir}/onboarding-device-$(get_device_guid).log"
+  rm -f "${onboarding_log}"
+  set +e  # temporarily disable exit on failure, we expect onboard to fail here:
+  run_fido_device_onboard --debug --wget-dir "${wget_device_download_dir}"
+  rc=$?
+  set -e
+  if [ "${rc}" == "0" ]; then
+    echo "❌ Expected device onboard to fail!"
+    return 1
+  fi
+
+  # verify that the wget FSIM error is logged
+  if ! grep -q "error handling device service info .*fdo\.wget:error" "${onboarding_log}" ; then
+    echo "❌ Client failed to report FSIM error!"
+    return 1
+  fi
```

Comment:
As we already search in logs to check if the onboarding went well, would it make sense to write a function let's say `find_in_log_or_fail $log $string_to_find`?

Reply:
Sorry - missed that. I'll update the PR...